### PR TITLE
ENH: use codecov + templates for PRs and issues

### PR DESCRIPTION
**Pull request by yarikoptic**
codecov is quite neat -- in contrast to coveralls provides browser extensions so you could see what covered in code right by browsing github website.

Templates are new feature, finally, and here you could find pointers to already existing ones for more ideas and improvements:  https://github.com/devspace/awesome-github-templates

### CODE REVIEW

**Comment by yarikoptic** [comment]
I wonder why there is no coveralls report within checks... interesting ;)  but I would like to merge quickly so we could see if actually codecov is as cool as I think it is ;)

**Comment by yarikoptic** [comment]
@fail2ban/contributors @fail2ban/developers -- any recommendations on may be what else to ask for in those templates?

**Comment by szepeviktor** [comment]
It is already too hard to fill in.

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+_Please fill out following information while also paying attention to comments.
+Depending on your issue not all fields might apply -- just remove those which
+do not apply, and remove the comments.  Thanks!_
+
+### Environment:
+
+#### Your OS (including release name/version):
```
Comment (szepeviktor) [inline .github/ISSUE_TEMPLATE.md]: No space after OS line.

**Comment by sebres** [comment]
I missed the whole thing here. 
I've commited an issue template direct in master, so will now revert to prev revision...

Here is my version if there is interest:

##### Actual and expected behavior

##### Steps to reproduce the problem

##### Configuration, dump and another helpful excerpts
```
```
##### Version of Fail2Ban and OS


---
Makes sure these boxes are checked before submitting your issue - thank you!
If not and you however want to post the issue, replace a `x` in checkbox ([x] through [ ]).

- [x] The version of Fail2Ban >= 0.9 (or verified the issue with this version also)
- [x] No foreign patchs installed and no customization made (the issue checked without all that)
- [x] The log file "fail2ban.log" contains no information to resolve the issue (or not understandable)

<sup> -- read and remove it --
We will be very grateful, if your problem will be described as completely as possible, encloses the log excerpts (if possible within DEBUG mode, if no errors evident within INFO mode), an excerpt of affected configuration resp. dumps of your configuration if expected (for example with ` fail2ban-client -d | grep 'affected-jail-name' `). 
Thank you in advance, because such issues like "It does not work" can not help us at all to resolve it!
-- / -- </sup>

**Comment by yarikoptic** [comment]
Ok -- adjusted issue template melding it with a version from @sebres .  You can already see a note on codecov coverage.  In another project where I enabled it, diffs look quite more usable, so it is all win/win.  If no further comments, I would merge it tomorrow.

**Pull request merged**